cilium/datadog_checks/cilium/cilium.py

```diff
@@ -0,0 +1,124 @@
+    'cilium_kubernetes_events_total': 'kubernetes.events.total',
+    'cilium_nodes_all_datapath_validations_total': 'nodes.all_datapath_validations.total',
+    'cilium_nodes_all_events_received_total': 'nodes.all_events_received.total',
+    'cilium_nodes_all_num': 'nodes.managed.total',
+    'cilium_policy_count': 'policy.count',
+    'cilium_policy_endpoint_enforcement_status': 'policy.endpoint_enforcement_status',  # double check
+    'cilium_policy_import_errors': 'policy.import_errors.count',
+    'cilium_policy_l7_denied_total': 'policy.l7_denied.total',
+    'cilium_policy_l7_forwarded_total': 'policy.l7_forwarded.total',
+    'cilium_policy_l7_parse_errors_total': 'policy.l7_parse_errors.total',
+    'cilium_policy_l7_received_total': 'policy.l7_received.total',
+    'cilium_policy_max_revision': 'policy.max_revision',
+    'cilium_policy_regeneration_time_stats_seconds': 'policy.regeneration_time_stats.seconds',
+    'cilium_policy_regeneration_total': 'policy.regeneration.total',
+    'cilium_process_cpu_seconds_total': 'process.cpu.seconds.total',
+    'cilium_process_max_fds': 'process.max_fds',
+    'cilium_process_open_fds': 'process.open_fds',
+    'cilium_process_resident_memory_bytes': 'process.resident_memory.bytes',
+    'cilium_process_start_time_seconds': 'process.start_time.seconds',
+    'cilium_process_virtual_memory_bytes': 'process.virtual_memory.bytes',
+    'cilium_process_virtual_memory_max_bytes': 'process.virtual_memory.max.bytes',
+    'cilium_subprocess_start_total': 'subprocess.start.total',
+    'cilium_triggers_policy_update_call_duration_seconds': 'triggers_policy.update_call_duration.seconds',
+    'cilium_triggers_policy_update_folds': 'triggers_policy.update_folds',
+    'cilium_triggers_policy_update_total': 'triggers_policy.update.total',
+    'cilium_unreachable_health_endpoints': 'unreachable.health_endpoints',
+    'cilium_unreachable_nodes': 'unreachable.nodes',
+    'cilium_event_ts': 'event_timestamp',
+}
+
+OPERATOR_METRICS = {
+    'cilium_operator_process_cpu_seconds_total': 'operator.process.cpu.seconds',
+    'cilium_operator_process_max_fds': 'operator.process.max_fds',
+    'cilium_operator_process_open_fds': 'operator.process.open_fds',
+    'cilium_operator_process_resident_memory_bytes': 'operator.process.resident_memory.bytes',
+    'cilium_operator_process_start_time_seconds': 'operator.process.start_time.seconds',
+    'cilium_operator_process_virtual_memory_bytes': 'operator.process.virtual_memory.bytes',
+    'cilium_operator_process_virtual_memory_max_bytes': 'operator.process.virtual_memory_max.bytes',
+    # TODO: ENI metrics are not listed
+}
+
+
+class CiliumCheck(OpenMetricsBaseCheck):
+    """
+    Collect Cilium metrics from Prometheus endpoint
+    """
+
+    def __init__(self, name, init_config, instances):
+        super(CiliumCheck, self).__init__(name, init_config, instances=self._generate_instances(instances))
+
+    def _generate_instances(self, instances):
```

Comment:
```suggestion
    def _generate_instances(self):
```
No need to pass instances, as instances is a one element list and you can access that element with `self.instance`